test/unit/puppetlabs/puppetserver/ruby/http_client_test.clj

```diff
@@ -181,166 +219,164 @@ jruby-config :- jruby-schemas/JRubyConfig
         (with-webserver-with-protocols nil nil
-          (let [sc (create-scripting-container 10080 {:use-ssl true})]
-            (.runScriptlet sc "$response = $c.get('/', {})")
-            (is (= "200" (.runScriptlet sc "$response.code")))
-            (is (= "hi" (.runScriptlet sc "$response.body")))
-            (.runScriptlet sc (str (format "$c = Puppet::Server::HttpClient.new('localhost', %d, {:use_ssl => %s});"
-                                           port false)
-                                   "$response = $c.get('/', {})"))
-            (is (= "200" (.runScriptlet sc "$response.code")))
-            (is (= "bye" (.runScriptlet sc "$response.body")))
-            (terminate-scripting-container sc))))))
+          (with-scripting-container sc
+            (with-http-client sc 10080 {:use-ssl true}
+              (.runScriptlet sc "$response = $c.get('/', {})")
+              (is (= "200" (.runScriptlet sc "$response.code")))
+              (is (= "hi" (.runScriptlet sc "$response.body")))
+              (.runScriptlet sc (str (format "$c = Puppet::Server::HttpClient.new('localhost', %d, {:use_ssl => %s});"
+                                            port false)
+                                    "$response = $c.get('/', {})"))
+              (is (= "200" (.runScriptlet sc "$response.code")))
+              (is (= "bye" (.runScriptlet sc "$response.body")))))))))
 
   (testing "can make https calls after http calls without a new scripting container"
     (logutils/with-test-logging
       (jetty9/with-test-webserver ring-app-alternate port
         (with-webserver-with-protocols nil nil
-          (let [sc (create-scripting-container port)]
-            (.runScriptlet sc "$response = $c.get('/', {})")
-            (is (= "200" (.runScriptlet sc "$response.code")))
-            (is (= "bye" (.runScriptlet sc "$response.body")))
-            (.runScriptlet sc (str (format "$c = Puppet::Server::HttpClient.new('localhost', %d, {:use_ssl => %s});"
-                                           10080 true)
-                                   "$response = $c.get('/', {})"))
-            (is (= "200" (.runScriptlet sc "$response.code")))
-            (is (= "hi" (.runScriptlet sc "$response.body")))
-            (terminate-scripting-container sc)))))))
+          (with-scripting-container sc
+            (with-http-client sc port {:use-ssl false}
+              (.runScriptlet sc "$response = $c.get('/', {})")
+              (is (= "200" (.runScriptlet sc "$response.code")))
+              (is (= "bye" (.runScriptlet sc "$response.body")))
+              (.runScriptlet sc (str (format "$c = Puppet::Server::HttpClient.new('localhost', %d, {:use_ssl => %s});"
+                                             10080 true)
+                                    "$response = $c.get('/', {})"))
+              (is (= "200" (.runScriptlet sc "$response.code")))
+              (is (= "hi" (.runScriptlet sc "$response.body"))))))))))
```

Comment:
This commit seems fine, except possibly exploring doing something a little simpler than the macro/dynamic var/binding stuff, but it's not a blocker or anything.